File: api/src/main/java/fr/huiitre/tools/modules/riot/valorant/application/catalog/view/ValorantNightMarketOffer.java
package fr.huiitre.tools.modules.riot.valorant.application.catalog.view;

import fr.huiitre.tools.modules.riot.valorant.application.skin.view.ValorantSkinView;

public record ValorantNightMarketOffer(
    String offerId,
    ValorantSkinView skin,
    int originalCost,
    int discountedCost,
    int discountPercent,
    boolean isSeen
) {}

File: api/Modules/Riot/Valorant/Application/Catalog/Views/ValorantNightMarketOffer.cs
using Tools.Api.Modules.Riot.Valorant.Application.Skin.Views;

namespace Tools.Api.Modules.Riot.Valorant.Application.Catalog.Views;

public sealed record ValorantNightMarketOffer(
    string OfferId,
    ValorantSkinView Skin,
    int OriginalCost,
    int DiscountedCost,
    int DiscountPercent,
    bool IsSeen
);